### ISSUE TITLE
ERROR: lremovexattr security.selinux failed: Permission denied

### ISSUE THREAD

**Issue by RomuloPBenedetti** [digint/btrbk#160]
Any Idea why I'm getting this?

```
Creating backup: /mnt/HDDs/backups/snapshots/desktop/root.20170630T0901
  35MiB 0:00:13 [ 2,6MiB/s] [ 2,6MiB/s]                                                                                                                                                                                                
WARNING: [send/receive] (send=/mnt/FS_TREE/snapshots/root.20170630T0901, receive=/mnt/HDDs/backups/snapshots/desktop) At subvol /mnt/FS_TREE/snapshots/root.20170630T0901
ERROR: lremovexattr home/rbenedetti/.config/vivaldi/Default/IndexedDB/https_www.netflix.com_0.indexeddb.leveldb/000397.log security.selinux failed: Permission denied
WARNING: [send/receive] (send=/mnt/FS_TREE/snapshots/root.20170630T0901, receive=/mnt/HDDs/backups/snapshots/desktop) At snapshot root.20170630T0901
ERROR: Failed to send/receive btrfs subvolume: /mnt/FS_TREE/snapshots/root.20170630T0901 [/mnt/FS_TREE/snapshots/root.20170630T0001] -> /mnt/HDDs/backups/snapshots/desktop
WARNING: Deleted partially received (garbled) subvolume: /mnt/HDDs/backups/snapshots/desktop/root.20170630T0901
ERROR: Error while resuming backups, aborting

```

**Comment by digint** [comment]
You are getting this because you removed the `security.selinux` xattr on the file `[...]/000397.log`. The btrfs (incremental) send stream wants to "replay" the xattr remove on the target side, but fails with "Permission denied" ([cmds-receive.c, L910](https://github.com/kdave/btrfs-progs/blob/master/cmds-receive.c#L910)).

Now I'm no expert on selinux, but it seems that you have enabled selinux on the target side, and some policy does not like you to delete "security.selinux".

**Comment by Szpadel** [comment]
It would be great if someone could provide selinux config to solve this.
Because of it btrbk is unusable on selinux enabled systems

**Comment by xenithorb** [comment]
Hello fellow bug members, if you have anything to contribute to the ^ upstream bug, please do so now. One thing that might help is in figuring out why btrfs send even wants to execute a xattr removal in the first place since we know that those attributes can't be removed on SELinux permissive and above systems. If such a thing is possible, then how does it record that as a difference between incrementals?

**Comment by digint** [comment]
Thanks @xenithorb for tracking this down: **upstream bug** to: kdave/btrfs-progs#68

Looks like there's no easy workaround for this yet.

**Comment by xenithorb** [comment]
Temporary (and kind of tedious) workaround here: https://github.com/kdave/btrfs-progs/issues/68#issuecomment-332279684

**Comment by mdonoughe** [comment]
This works for me. If btrbk fails to send an incremental backup, you can manually send a full backup for the snapshot that failed to send, and then from there btrbk will be able to continue from there.

    sudo btrfs send <source>/_btrbk_snap/<failed snapshot> | /usr/bin/pv -trab | sudo /usr/bin/ssh -i /etc/btrbk/id_rsa root@<other machine> btrfs receive -vv <destination>

I don't remember how I arrived at this command. I think it's based on what btrbk does.

The problem seems to be that during some sort of upgrade the file gets retagged and then btrfs tries to remove the old tag before setting the new tag. For me this only happens occasionally so I'm able to continue by executing this workaround once every few months or so when btrbk stops working.

**Comment by xenithorb** [comment]
Yeah it would appear that this does work, however the downside is that you need the space available in order to accommodate the full backup in its entirety. Whereas the workaround I suggested in kdave/btrfs-progs#68 takes far longer and is a bit convoluted, but allows you to conserve space by continuing the incremental chain.

**Comment by alterNERDtive** [comment]
My current lazy AF workaround is doing `ssh -t <backuphost> sudo setenforce 0 && btrbk resume; ssh -t <backuphost> sudo setenforce 1` …

Anyone know if there’s a cleaner way? E.g. disabling selinux enforcement for a specific mount? Google-Fu failed me there, sadly.
